### PR TITLE
refactor: Migrate grouped-conversations to react

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/components/list/GroupedConversations.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  ConversationLabel,
+  createLabel,
+  createLabelFavorites,
+  createLabelGroups,
+  createLabelPeople,
+} from '../../conversation/ConversationLabelRepository';
+import {generateConversationUrl} from '../../router/routeGenerator';
+import {createNavigate} from '../../router/routerBindings';
+
+import './GroupedConversationHeader';
```
Comment (AndyLnd) [inline src/script/components/list/GroupedConversations.tsx]: This can go, right?

File: src/script/components/list/GroupedConversations.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      .filter(({conversations}) => !!conversations().length);
+    folders.push(...custom);
+
+    return folders;
+  })();
+
+  const isExpanded = (folderId: string): boolean => expandedFolders.includes(folderId);
+  const toggle = (folderId: string): void => {
+    if (isExpanded(folderId)) {
+      setExpandedFolders(expandedFolders.filter(folder => folder !== folderId));
```
Comment (AndyLnd) [inline src/script/components/list/GroupedConversations.tsx:118]: As these changes rely on the previous state, a setter function would be more fitting: 
```suggestion
      setExpandedFolders(expanded => expanded.filter(folder => folder !== folderId));
```

File: src/script/components/list/GroupedConversations.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    } else {
+      setExpandedFolders([...expandedFolders, folderId]);
```
Comment (AndyLnd) [inline src/script/components/list/GroupedConversations.tsx:120]: Same as above:
```suggestion
      setExpandedFolders(expanded => [...expanded, folderId]);
```

File: src/script/components/list/GroupedConversations.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  isVisibleFunc,
+  listViewModel,
+  onJoinCall,
+  setExpandedFolders,
+}) => {
+  const {conversationLabelRepository} = conversationRepository;
+  const makeOnClick = (conversationId: string, domain: string | null) =>
+    createNavigate(generateConversationUrl(conversationId, domain));
+
+  const folders = (() => {
```
Comment (AndyLnd) [inline src/script/components/list/GroupedConversations.tsx]: This looks pretty expensive. Would a useMemo be appropriate here?

**Review by AndyLnd — CHANGES_REQUESTED** [review]

File: src/script/components/list/GroupedConversations.tsx (hunk 1 above)
Comment (thisisamir98) [reply src/script/components/list/GroupedConversations.tsx]: https://github.com/wireapp/wire-webapp/pull/11621/commits/42b92f2df802505f110ebebad38a3dc30b46c3f0

File: src/script/components/list/GroupedConversations.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+import {generateConversationUrl} from '../../router/routeGenerator';
+import {createNavigate} from '../../router/routerBindings';
+
+import GroupedConversationHeader from './GroupedConversationHeader';
+import ConversationListCell from './ConversationListCell';
+import {ListViewModel} from 'src/script/view_model/ListViewModel';
+import {registerReactComponent, useKoSubscribableChildren} from 'Util/ComponentUtil';
+
+const useLabels = (conversationLabelRepository: ConversationLabelRepository) => {
+  const {labels: conversationLables} = useKoSubscribableChildren(conversationLabelRepository, ['labels']);
```
Comment (AndyLnd) [inline src/script/components/list/GroupedConversations.tsx]: typo :) 
`conversationLables` -> `conversationLabels`

File: src/script/components/list/GroupedConversations.tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  const [labels, setLabels] = useState<ConversationLabel[]>(conversationLables);
+
+  useEffect(() => {
+    if (!conversationLables) {
+      return setLabels([]);
+    }
+    const updateLabels = () => {
+      setLabels([...conversationLabelRepository.labels()]);
```
Comment (AndyLnd) [inline src/script/components/list/GroupedConversations.tsx]:
```suggestion
      setLabels([...conversationLabels]);
```

**Review by AndyLnd — CHANGES_REQUESTED** [review]

**Review by AndyLnd — APPROVED** [review]

**Pull request merged**